goconvey.go

```diff
@@ -45,15 +46,66 @@ func flags() {
 	flag.StringVar(&excludedDirs, "excludedDirs", "vendor,node_modules", "A comma separated list of directories that will be excluded from being watched.")
 	flag.StringVar(&workDir, "workDir", "", "set goconvey working directory (default current directory).")
 	flag.BoolVar(&autoLaunchBrowser, "launchBrowser", true, "toggle auto launching of browser.")
+	flag.BoolVar(&cleanupTmpFolder, "cleanupTemp", false, "cleanup tmp folder with reports and static web files")
 
 	log.SetOutput(os.Stdout)
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 }
 func folders() {
 	_, file, _, _ := runtime.Caller(0)
 	here := filepath.Dir(file)
-	static = filepath.Join(here, "/web/client")
-	reports = filepath.Join(static, "reports")
+	localStatic := filepath.Join(here, "web", "client")
+
+	currentUser, err := user.Current()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	tmpDest := filepath.Join(os.TempDir(), fmt.Sprintf("goconvey-%v", currentUser.Uid))
+
+	mkdir := func(folder string) {
+		if err := os.MkdirAll(folder, 0700); err != nil {
+			if !os.IsExist(err) {
+				log.Fatal(err)
+			}
+		}
+	}
+
+	cleanupAndMkdir := func(folder string) {
+		if stat, err := os.Stat(folder); err == nil && stat != nil {
+			if !stat.IsDir() {
+				if err = os.Remove(folder); err != nil {
+					log.Fatal(err)
+				}
+
+				mkdir(folder)
+			} else {
+				if cleanupTmpFolder {
+					if err = os.RemoveAll(folder); err != nil {
+						log.Fatal(err)
+					}
+
+					mkdir(folder)
+				}
+			}
+		} else {
+			if err != nil && stat == nil {
+				if os.IsNotExist(err) {
+					mkdir(folder)
+				} else {
+					log.Fatal(err)
+				}
+			}
+		}
+	}
+
+	tmpReportsPath := filepath.Join(tmpDest, "reports")
```

Comment:
> Can't tmpDest just be the location of the reports directly?

Just in case we'll need to put there some other outputs, besides reports.

> which can be served from the /reports/ endpoint of the built-in http server?

Yes.

Reply:
> Just in case we'll need to put there some other outputs

Ah, future proofing of sorts; IMO since all this temp folder business is very ephemeral (new goconvey process blows away the whole thing), then I'd be inclined to not worry about that and it could be added later if needed without much concern.

Reply:
>  then I'd be inclined to not worry about that and it could be added later if needed without much concern.

I also thought about adding a race detector and other stuff, mutational testing etc to Goconvey...
After fixing goconvey/goroutine business you've mentioned, and improving coverage to a more or less meaningful value.

Reply:
Yeah, those would all be pretty good improvements; Maybe rename the 'reports' to 'coverage_out' so it's clearer what it's for, and add a comment that it's a subfolder to allow future expansion of outputs?

Reply:
(don't have to change the http route for now; that can stay as 'reports')

Reply:
Will do